src/otx/core/model/entity/base.py

```diff
@@ -328,7 +339,7 @@ class OVModel(OTXModel, Generic[T_OTXBatchDataEntity, T_OTXBatchPredEntity, T_OT
     def __init__(
         self,
         num_classes: int,
-        model_name: str,
+        model_name: str | dict[str, str],
```

Comment:
Why is this change needed?

Reply:
Due to `_model_name` set for visual prompting models.
https://github.com/openvinotoolkit/training_extensions/blob/2708d7ba16465e35cd7665da3043223ecc2dde3f/src/otx/core/model/entity/visual_prompting.py#L192-L204

Reply:
I can see there is a pre-defined naming rule here
https://github.com/openvinotoolkit/training_extensions/blob/2708d7ba16465e35cd7665da3043223ecc2dde3f/src/otx/core/model/entity/visual_prompting.py#L192-L195
Is there any problem with just accepting `model_name` here, then change this
https://github.com/openvinotoolkit/training_extensions/blob/2708d7ba16465e35cd7665da3043223ecc2dde3f/src/otx/core/model/entity/visual_prompting.py#L210
to
```python
        model_names = {
            module: model_name.replace(basename, f"visual_prompting_{module}.xml")
            for module in ["image_encoder", "decoder"]
        }
```

Reply:
Updated at https://github.com/openvinotoolkit/training_extensions/pull/2955/commits/e521fd7ec97c6285ee2b4ef7a261c8d5ffdc1a78.